test_requests.py

```diff
@@ -188,6 +188,26 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self, httpbin):
         assert r.history[0].status_code == 302
         assert r.history[0].is_redirect
 
+    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
+        try:
+            requests.get(httpbin('redirect', '50'))
+        except TooManyRedirects as e:
+            assert e.request is not None
+            assert len(e.response.history) == DEFAULT_REDIRECT_LIMIT
+        else:
+            assert False
```

Comment:
Rather than using `assert False`, let's use [`pytest.fail`](https://pytest.org/latest/example/simple.html#writing-well-integrated-assertion-helpers) so that we can have a convenient message telling us what went wrong.